File: apps/ios/SmartAssist/Sources/Services/DeviceStore.swift
import Foundation

public actor DeviceStore {
    private var devices: [Device] = []

    public init() {}

    public func allDevices() -> [Device] {
        devices
    }

    public func add(_ device: Device) {
        devices.append(device)
    }

    public func remove(id: String) {
        devices.removeAll { $0.id == id }
    }

    public func updateConnection(id: String, connected: Bool) {
        if let index = devices.firstIndex(where: { $0.id == id }) {
            devices[index].connected = connected
        }
    }
}

File: apps/android/app/src/main/java/com/smartassist/data/DeviceStore.kt
package com.smartassist.data

class DeviceStore {
    private val devices = mutableListOf<Device>()

    fun allDevices(): List<Device> = devices.toList()

    fun add(device: Device) {
        devices.add(device)
    }

    fun remove(id: String) {
        devices.removeAll { it.id == id }
    }

    fun updateConnection(id: String, connected: Boolean) {
        val index = devices.indexOfFirst { it.id == id }
        if (index != -1) {
            devices[index] = devices[index].copy(connected = connected)
        }
    }
}
